datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateDescriptionResolver.java

```diff
@@ -135,6 +139,62 @@ private CompletableFuture<Boolean> updateDatasetDescription(Urn targetUrn, Descr
     });
   }
 
+  private CompletableFuture<Boolean> updateChartDescription(Urn targetUrn, DescriptionUpdateInput input, QueryContext context) {
+
+    return CompletableFuture.supplyAsync(() -> {
+
+      if (!DescriptionUtils.isAuthorizedToUpdateFieldDescription(context, targetUrn)) {
+        throw new AuthorizationException(
+            "Unauthorized to perform this action. Please contact your DataHub administrator.");
+      }
+
+      if (input.getSubResourceType() == null) {
+        throw new IllegalArgumentException("Update description without subresource is not currently supported");
+      }
```

Comment:
ah it took me looking at how we currently update chart descriptions to realize this isn't a problem. It is pretty weird that we use `updateChart` to update chart's description but use `updateDescription` to update its subresource descriptions.

Also looking at it more we're doing the same thing for datasets and their subresources.

I believe it's out of the scope of this PR but that might be something we want to consolidate in the future - we could check if it has a subresource in the input or not and update one thing or another